analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarReportingContextBase.cs

```diff
@@ -77,6 +77,11 @@ private static Diagnostic EnsureDiagnosticLocation(Diagnostic diagnostic)
             diagnostic.AdditionalLocations.Select(l => l.EnsureMappedLocation()).ToImmutableList(),
             diagnostic.Properties);
     }
+
+    private bool CanDiagnosticBeReported(Diagnostic diagnostic) =>
```

Comment:
Since it is invoked only in 1 place, why did you create a separate method for it?